### PR TITLE
Stopat

### PR DESCRIPTION
**Pull request by Stuart-Moore**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [x] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [x] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [x] Documentation
 - [ ] Build system
 

### Purpose
Thought I sorted restoring with StopAtMark stuff years ago. Turns out I hadn't, so I have 

### Commands to test
Can now do this:

Restore-DbaDatabase -SqlInstance server1\instance -Path \\Server\files$\db1 -StopMark ImportantTransaction -StopAfterDate (Get-Date).AddHours(-12)

Which will restore to the first transaction marked as 'ImportantTransaction' after 12 hours ago

Good job I checked this existed before I published the post about it. psm1 isn't an edited file, it went missing so I put back the one from dev and git has just marked it as new.

### CODE REVIEW

**Comment by Stuart-Moore** [comment]
Failed branch test was Invoke-DbaDbClone which I've not touched

**Comment by wsmelton** [comment]
Yeah, it looks like Appveyor is 💩 out on that test for some reason...I have some work to take care of but will try to review this one later "my" afternoon 😉

**Comment by Stuart-Moore** [comment]
Thanks, nice to know it's not me for once.

Just snuck in a switch parameter that was already plumbed in internally just not publically, -IgnoreDiffBackups

**Comment by wsmelton** [comment]
@Stuart-Moore can you revert the psm1 file from your branch? I'm not all that sure how it showed up.

File: functions/Invoke-DbaAdvancedRestore.ps1 [hunk 1]
```diff
@@ -210,7 +222,19 @@ function Invoke-DbaAdvancedRestore {
                 } else {
                     $Restore.NoRecovery = $False
                 }
-                if ($restoretime -gt (Get-Date) -or $Restore.RestoreTime -gt (Get-Date) -or $backup.RecoveryModel -eq 'Simple') {
+                if ('' -ne $StopMark) {
```
[the remark this answers is not in the document]
Comment (wsmelton) [reply functions/Invoke-DbaAdvancedRestore.ps1:225]: Few comments...

**Comment by wsmelton** [comment]
Stupid git configuration, forgot I had to update my local git config for work...so it picked up the other account I use....maybe when we squash this PR it will ignore it in our git history :(

FYI, as long as you have Git 2.23 or higher you can now "restore" a file that gets corrupt or messed up like the psm1 file seems to do commonly...

I simply ran this command to restore it from development, with the branch of this PR checked out:

```
git restore -s development dbatools.psm1
```

File: functions/Restore-DbaDatabase.ps1 [hunk 2]
```diff
@@ -82,6 +82,9 @@ function Restore-DbaDatabase {
     .PARAMETER IgnoreLogBackup
         This switch tells the function to ignore transaction log backups. The process will restore to the latest full or differential backup point only
 
+    .PARAMETER IgnoreDiffBackup
+        This switch tells the function to ignore differential log backups. The process will restore to the latest full and onwards with transaction log backups only
```
[the remark this answers is not in the document]
Comment (Stuart-Moore) [reply functions/Restore-DbaDatabase.ps1]: Fixed

File: functions/Restore-DbaDatabase.ps1 [hunk 3]
```diff
@@ -325,6 +338,10 @@ function Restore-DbaDatabase {
 
         Restores 'database' to 'server1' and moves the files to new locations. The format for the $FileStructure HashTable is the file logical name as the Key, and the new location as the Value.
 
+    .EXAMPLE
+        PS C:\> Restore-DbaDatabase -SqlInstance server1 -Path \\ServerName\ShareName\File -DatabaseName database -DatabaseName database -StopMark OvernightStart -StopBefore -StopAfterDate Get-Date('21:00 10/05/2020')
+
+        Restores the backups from \\ServerName\ShareName\File as database, stops before the first 'OvernightStop' mark that occurs after '21:00 10/05/2020'
```
[the remark this answers is not in the document]
Comment (Stuart-Moore) [reply functions/Restore-DbaDatabase.ps1]: Added a note

File: functions/Invoke-DbaAdvancedRestore.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (Stuart-Moore) [reply functions/Invoke-DbaAdvancedRestore.ps1]: Done

**Comment by Stuart-Moore** [comment]
I did exactly the same yesterday with a new dev box at work.

**Comment by Stuart-Moore** [comment]
@wsmelton are you happy with the changes? If you are I can merge.

Will stop the 33 -ne 28 errors in the appveyor tests if nothing else ;)

**Pull request merged**